api/src/main/java/org/openmrs/module/fhir2/providers/r4/PatientFhirResourceProvider.java

```diff
@@ -143,4 +146,18 @@ public IBundleProvider searchPatients(@OptionalParam(name = Patient.SP_NAME) Str
 		return patientService.searchForPatients(name, given, family, identifier, gender, birthDate, deathDate, deceased,
 		    city, state, postalCode, country, id, lastUpdated, sort, revIncludes);
 	}
+
+	/**
+	 * The $everything operation fetches all the information related the specified patient
+	 * @param patientId The id of the patient
+	 * @return a bundle of resources which reference to or are referenced from the patient
+	 */
+	@Operation(name = "everything", idempotent = true, type = Patient.class, bundleType = BundleTypeEnum.SEARCHSET)
+	public IBundleProvider getPatientEverything(@IdParam IdType patientId) {
```

Comment:
@varung-31 @ibacher PTAL.

Reply:
@Medhavi-16 I'd do the `null` check something like this:

```java
if (patientId == null) {
    return null;
}
```

Reply:
PS With my other suggestions, this can then be simplified to:

```java
TokenParam patientReference = new TokenParam(patientId.getIdPart());
```

Reply:
@ibacher I think for the null check, `(id == null || id.getIdPart() == null)` should be better.

Reply:
Even better might be `(id == null || id.getIdPart() == null || id.getIdPart().isEmpty())`

Reply:
I have a small question here: How do we actually supply null or empty values in a path variable?
Because trying to hit the `https://hapi.fhir.org/baseR4/Patient//$everything` URL redirects me to the type-level $everything operation instead.

Reply:
So I don't think there's an explicit way to craft a URL that does that or if there were, it would be an error somewhere in HAPI's code base. But if such an error is accidentally introduced at some point, I'd prefer we do something sensible here rather than send something far down the chain and, e.g., run a query to select all of the patients and everything associated with them.